Implemented Dashboard Service.

nmcmahon-unosquare opened the pull request:
Built out method implementations for service.
Added Contracts and Events to Employee for easy reference.
Populated EmployeeSnapshotDto
Add DashboardService to dependency injection.
Implemented some controller methods.
Added a hardcoded mock database instance to enable testing of complex database queries.
Added some words to spelling dictionary.
Added utility methods to date service.

Jamie-Unosquare commented on v2/AdminCore.Services/DashboardService.cs:
```diff
@@ -2,39 +2,173 @@
+      var startOfMonth = DateService.GetMonthStartDate(date);
+      var endOfMonth = DateService.GetMonthEndDate(date);
+
+      return (evnt.EmployeeId == employeeId) &&
```
Extract this into a more readable return method

Jamie-Unosquare commented on v2/AdminCore.Services/DashboardService.cs:
```diff
@@ -2,39 +2,173 @@
+
+    public static bool GetEmployeeSnapshotByEmployeeIdQuery(int employeeId, Contract contract, DateTime date)
+    {
+      return contract.EmployeeId == employeeId &&
```
Same for this and other returns

nmcmahon-unosquare replied:
GetEmployeeSnapshotByEmployeeIdQuery is the extracted method

Jamie-Unosquare commented on v2/AdminCore.Common/Interfaces/IDashboardService.cs:
```diff
@@ -9,7 +9,7 @@ public interface IDashboardService
     IList<EventDto> GetEmployeeDashboardEvents(int employeeId, DateTime date);
 
-    IList<EventDto> GetEmployeeEvents(int employeeId, DateTime date);
+    IList<EventDto> GetEmployeeEventsForMonth(int employeeId, DateTime date);
```
EmployeeId is no longer being passed into the Services, instead we're using AuthenticatedUser injection. You can see this in the HolidayController/EventService

nmcmahon-unosquare replied:
Authenticated user is being injected by the controller, employeeId is included in this service method for re-usability.

Jamie-Unosquare commented on v2/AdminCore.Services/DashboardService.cs:
```diff
@@ -2,39 +2,173 @@
+      return _mapper.Map<IList<EventDto>>(eventList);
+    }
+    
+    public static bool EmployeeDashboardEventsQuery(int employeeId, DateTime date, Event evnt)
```
Same as before, use employeeId when it's specifically querying for another employee, otherwise use the authenticatedUser.getUserId()

nmcmahon-unosquare replied:
This method is the query ran against the DB. It's called by the service. EmployeeId needs to be here

Jamie-Unosquare commented on v2/AdminCore.Services/DateService.cs:
```diff
@@ -22,5 +25,73 @@ public DateTime GetEndOfYearDate()
+          (DateIsBetweenRangeOfDates(startDateOfRange1, startDateOfRange2, endDateOfRange2) && endDateOfRange1 >= endDateOfRange2);
+        }
+      
+        public static bool ContractIsActiveDuringDate(Contract contract, DateTime date)
```
Why is this in the DateService and not ContractService?

nmcmahon-unosquare replied:
Chatted with Jamie. This is fine

Jamie-Unosquare reviewed (changes requested):
A few changes requested

nmcmahon-unosquare committed:
Implemented Dashboard Service.

Built out method implementations for service.
Added Contracts and Events to Employee for easy reference.
Populated EmployeeSnapshotDto
Add DashboardService to dependency injection.
Implemented some controller methods.
Added a hardcoded mock database instance to enable testing of complex database queries.
Added some words to spelling dictionary.
Added utility methods to date service.

nmcmahon-unosquare committed:
Change cors policy usage from services to app
